<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 66px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 71px;
  margin-top: -71px;
}

.section h2 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h3 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h4 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h5 {
  padding-top: 71px;
  margin-top: -71px;
}
.section h6 {
  padding-top: 71px;
  margin-top: -71px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Population genetics and data analysis workshop in R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="organizers.html">Meet the organizers</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Core Concepts</li>
    <li>
      <a href="Core_Concepts.html"></a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Installing R and Rstudio</li>
    <li>
      <a href="installR.html"></a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Day3
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Microsatellite data analysis</li>
    <li>
      <a href="ssr.html"></a>
    </li>
    <li class="dropdown-header">SNP data analysis</li>
    <li>
      <a href="SNP.html"></a>
    </li>
  </ul>
</li>
<li>
  <a href="participants.html">List of participants</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="section" class="section level1">
<h1></h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Population genetics is the study of genetic composition of the population and forces that cause changes in this composition. Simply, it is also the study of forces like mutation, migration, selection and drift. Before diving into details, there are some key concepts that we need to learn.</p>
<p><strong>Allele:</strong> It is a variant form of a gene e.g. Avr3a gene in plant pathogen Phytophthora infestans have either E or K amino acid at position xx and either M or I at position xx. Isolates which have KI genotype initiate a hypersensitive response in R3a plants where isolates with EM genotype evade this recognition and cause disease on R3a plants. You can also say that, population genetics is about the alleles in the populations.</p>
<p><strong>Locus (plural loci):</strong> Position in a genome where one or more alleles are segregating.</p>
<p><strong>Genotype:</strong> Genotype is the combination of the alleles at a particular locus.</p>
<p><strong>Di-allelic model:</strong> Model which assumes that there is only two alleles.</p>
<p>If an individual has same allele at a locus, we call that <strong>homozyogous individua</strong>l e.g AA or aa If an individual has different allele at a locus, we call that <strong>heterozygous individual</strong> e.g Aa</p>
<p><strong>Allele frequency:</strong> Frequency of copies of the allele in the population divided by total number of gene copies in the population.</p>
<p><strong>Genotype frequency:</strong> Number of individuals of certain genotype divided by total number of individuals.</p>
<p><strong>e.g. Lets look at a locus that carries two alleles, A and B. In diploids, we would observed three possible genotypes, two homozygous (AA and BB), and one heterozygous (AB). Lets say we found following genotype count, AA=38, AB=16, BB = 46. Calculate genotype frequency and allele frequency.</strong></p>
</div>
<div id="first-lets-look-at-genotype-freq" class="section level2">
<h2>first lets look at genotype freq</h2>
<pre class="r"><code>AA &lt;-  38
AB &lt;- 16
BB &lt;- 46
N &lt;- AA+AB+BB
  
AA_freq &lt;- AA/N
AA_freq</code></pre>
<pre><code>## [1] 0.38</code></pre>
<pre class="r"><code>BB_freq &lt;- BB/N
BB_freq</code></pre>
<pre><code>## [1] 0.46</code></pre>
<pre class="r"><code>AB_freq &lt;- AB/N
AB_freq</code></pre>
<pre><code>## [1] 0.16</code></pre>
</div>
<div id="now-lets-calculate-allele-freq" class="section level2">
<h2>Now lets calculate allele freq</h2>
<p>remember allele freq is frequency of copies of the allele in the population divided by total number of gene copies in the population.</p>
<p>For allele A and B</p>
<pre class="r"><code>A &lt;- 2*38 + 16
B &lt;- 2*46 + 16
Gene_copies &lt;- 200 ## why is this 200

allele_freq_A &lt;- A/Gene_copies
allele_freq_B &lt;- B/Gene_copies

allele_freq_A</code></pre>
<pre><code>## [1] 0.46</code></pre>
<pre class="r"><code>allele_freq_B</code></pre>
<pre><code>## [1] 0.54</code></pre>
</div>
<div id="write-a-simple-function-to-calculate-genotype-and-allele-frequencies." class="section level2">
<h2>Write a simple function to calculate genotype and allele frequencies.</h2>
<p>Remember, we said that population genetics is about studying the changes in the allele frequency. Now if you ask, how these allele frequencies are going to change overtime, then you are in right path.</p>
</div>
<div id="hardy-weinberg-equilibrium-hwe" class="section level2">
<h2>Hardy Weinberg equilibrium (HWE)</h2>
<p>Previously, we calculated allele frequencies given the genotype counts. Now the question is can we reverse calculate i.e can we predict genotype frequencies from allele frequencies. Yes, we can probably do this but we have to make few assumptions first.</p>
<p>Random mating: One individual mate with any other individual with the same probablity.</p>
<p>Lets consider a simple example of two alleles A and a whose allele freqencies are p and q respectively. Then, Pr(AA) = p2, Pr(aa) = q2 and P(Aa) = 2pq.</p>
<p>The Hardy–Weinberg principle, also known as the Hardy–Weinberg equilibrium, model, theorem, or law, states that allele and genotype frequencies in a population will remain constant from generation to generation in the absence of other evolutionary influences like mutation, migration, genetic drift, mate choice, assortative mating, natural selection, sexual selection, inbreeding etc.</p>
<p>Lets try to estimate genotype frequencies given the allele frequency.</p>
<pre class="r"><code>freq_A &lt;- 0.08
freq_a &lt;- 1-0.08

AA &lt;- freq_A * freq_A
aa &lt;- freq_a * freq_a

Aa &lt;- 2*freq_a*freq_A

AA</code></pre>
<pre><code>## [1] 0.0064</code></pre>
<pre class="r"><code>aa</code></pre>
<pre><code>## [1] 0.8464</code></pre>
<pre class="r"><code>Aa</code></pre>
<pre><code>## [1] 0.1472</code></pre>
<div id="molecular-markers" class="section level3">
<h3>Molecular markers</h3>
<div id="simple-sequence-repeats-or-microsatellites" class="section level4">
<h4>Simple Sequence Repeats or microsatellites</h4>
</div>
<div id="single-nucleotide-polymorphism-snps" class="section level4">
<h4>Single Nucleotide Polymorphism (SNPs)</h4>
</div>
</div>
<div id="population-structure" class="section level3">
<h3>Population structure</h3>
<div id="fst" class="section level4">
<h4>FST</h4>
</div>
<div id="clustering" class="section level4">
<h4>Clustering</h4>
</div>
<div id="minimum-spanning-network" class="section level4">
<h4>Minimum spanning network</h4>
</div>
</div>
<div id="discriminant-analysis-of-principle-components" class="section level3">
<h3>Discriminant analysis of principle components</h3>
</div>
<div id="section-1" class="section level3">
<h3></h3>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
